scipy/special/tests/test_zeta.py

```diff
@@ -47,3 +49,89 @@ def test_riemann_zeta_avoid_overflow():
+    #     cases.append((z, reference))
+
+    # # Moderately large imaginary part in each quadrant outside of critical strip
+    # for x_sign, y_sign in [1, 1], [1, -1], [-1, 1], [-1, -1]:
+    #     x = x_sign * rng.uniform(2, 8)
+    #     y = y_sign * rng.uniform(50, 80)
+    #     z = x + y*1j
+    #     reference = complex(mp.zeta(z))
+    #     cases.append((z, reference))
+
+    # # points in critical strip
+    # x = rng.uniform(-1.0, 1.0, size=5)
+    # y = np.exp(rng.uniform(0, 5, size=5))
+    # z = x + y*1j
+    # for t in z:
+    #     reference = complex(mp.zeta(t))
+    #     cases.append((complex(t), reference))
+    # z = x - y*1j
+    # for t in z:
+    #     reference = complex(mp.zeta(t))
+    #     cases.append((complex(t), reference))
+    [
+        # small values in each quadrant outside critical strip
+        ((3.12838509346655+7.111085974836645j),
+         (1.0192654793474945+0.08795174413289127j)),
+        ((7.06791362314716-7.219497492626728j),
+         (1.0020740683598117-0.006752725913243711j)),
+        ((-6.806227077655519+2.724411451005281j),
+         (0.06312488213559667-0.061641496333765956j)),
+        ((-3.0170751511621026-6.3686522550665945j),
+         (-0.10330747857150148-1.214541994832571j)),
+        # Moderately large imaginary part, outside critical strip
+        ((6.133994402212294+60.03091448000761j),
+         (0.9885701843417336+0.009636925981078128j)),
+        ((6.17268142822657-64.74883149743795j),
+         (1.0080474225840865+0.012032804974965354j)),
+        ((-3.462191939791879+76.16258975567534j),
+         (18672.072070850158+2908.5104826247184j)),
+        ((-6.955735216531752-74.75791554155748j),
+         (-77672258.72276545+71625206.0401107j)),
+        # Inside critical strip, varying imaginary part.
+        ((-0.1823923420352156+1.4596830498094384j),
+         (0.15641024400724224-0.29104726162518146j)),
+        ((0.9346987902419266+4.918968547259143j),
+         (0.7452215845376634+0.17593462255710995j)),
+        ((0.7384965359955509+66.6142398421354j),
+         (0.5365014352325163-0.35081725113825424j)),
+        ((-0.14456304559992472+21.783747851715468j),
+         (0.1373304916093494+1.6091267223996006j)),
+        ((-0.5904101064314209+33.17656449538932j),
+         (-4.409223214592868-1.6987333585451423j)),
+         ((-0.1823923420352156-1.4596830498094384j),
+          (0.15641024400724224+0.29104726162518146j)),
+        ((0.9346987902419266-4.918968547259143j),
+         (0.7452215845376634-0.17593462255710995j)),
+        ((0.7384965359955509-66.6142398421354j),
+         (0.5365014352325163+0.35081725113825424j)),
+        ((-0.14456304559992472-21.783747851715468j),
+         (0.1373304916093494-1.6091267223996006j)),
+        ((-0.5904101064314209-33.17656449538932j),
+         (-4.409223214592868+1.6987333585451423j)),
+    ]
+)
+def test_riemann_zeta_complex(z, desired):
+    assert_allclose(sc.zeta(z), desired, rtol=1e-13)
```

Comment:
Sometimes it's important to test real and imaginary parts separately (e.g. `logsumexp` but looks like it wouldn't make much of a difference here because the magnitudes tend to be similar.

The test LGTM (checked script and that reference values are produced by script). It might be worth testing a few other things:
- negative even integers (if those are not already tested) and surrounding points
- large real magnitude (around 49 ~ 51)
- what happens for _very_ large imaginary arguments, especially when Euler-Maclaurin is used? 

Would also be good to confirm what you documented about large imaginary component with real magnitude between 49 and 50.

Are there already tests for Hurwitz zeta (i.e. that it returns NaN for anything other than real $x \geq 1$)?

Reply:
Good ideas for more test cases.

 Hurwitz zeta not working for `x < 1` is just a limitation of the implementation, so not something that should be tested, except perhaps for tests we mark `xfail` with the intention of eventually fixing.

Reply:
My thought for why it might be OK not to test the real and imaginary parts separately is a heuristic that for an analytic function

$$\frac{|f(x + h) - f(x)|}{|h|}$$

should be roughly the same for $h$ sufficiently small in any direction, since the derivative will be the same in any direction.

In which case, if you were to do $g(\zeta(z))$ for some analytic $g$, then errors in `zeta(z).real` and `zeta(z).imag` of the same absolute size should contribute to the same error in $g(\zeta(z))$.

So as long as you restrict to analytic transformations, it's safe to only control relative error in complex space. But also, for methods which rely on analytic operations complex numbers, not the real part and imaginary individually, it's difficult to control the relative error for each component separately.

Reply:
> Hurwitz zeta not working for x < 1 is just a limitation of the implementation, so not something that should be tested

OK, I  just thought of it because you added documentation about it returning NaN.

Reply:
> OK, I just thought of it because you added documentation about it returning NaN.

I'll adjust the wording in the documentation. I was trying to document a current limitation, not a desired behavior.

Reply:
Oh, I just noticed that I had accidentally documented that `z` with `49 < z.real < 50` aren't supported for sufficiently large `z.imag`, but it actually should have been `-49 < z.real < 50`. That must have really confusing.

Reply:
Yeah, I realized that and figured it would get resolved with the other comments.